webapp/src/components/CompetenciesDetailTable.tsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import {
+  Box,
+  Collapse,
+  IconButton,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Rating,
+  Tooltip,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+import CircleOutlinedIcon from '@mui/icons-material/CircleOutlined';
+import CircleIcon from '@mui/icons-material/Circle';
+import { Competency, Reflection } from '../types/api';
+import useApiData from '../helpers/useApiData';
+import { formatDate } from '../helpers/dateTime';
+
+export interface IBackgroundColor {
+  [key: string]: string;
+}
+export const categoryBackgroundColor: IBackgroundColor = {
+  Functional: '#CAE2FA',
+  Strategic: '#ffe59a',
+  Operational: '#b6d7a8',
+  Behavioural: '#b4a7d5',
+  Organizational: '#F7B8D7',
+};
+
+const StyledRating = styled(Rating)({
+  '& .MuiRating-iconFilled': {
+    color: '#42a5f5',
+  },
+});
+
+interface IRatingValue {
+  [key: string]: number;
+}
+const ratingValue: IRatingValue = {
+  Aware: 1,
+  Novice: 2,
+  Intermediate: 3,
+  Advanced: 4,
+  Expert: 5,
+};
+
+interface DetailTableProps {
+  catgoryLabel: string;
+  categoryDescription: string;
+  competencies: Competency[];
+}
+
+const CompetenciesDetailTable = ({
+  catgoryLabel,
+  categoryDescription,
+  competencies,
+}: DetailTableProps) => {
+  const [open, setOpen] = React.useState(false);
```

Comment:
useState can directly be destructured in the import so that ``` useState()``` can be utilized directly instead of ```React.useState() ```

Reply:
Good point! I changed code to import `useState`